fabric8a_auth/auth.py

```diff
@@ -0,0 +1,208 @@
+        public_key = pub_key.get("key", "")
+
+        if audiences is None:
+            # Service account does not have audiences
+            try:
+                decoded_token = jwt.decode(token, public_key, algorithms=['RS256'])
+            except jwt.InvalidTokenError:
+                app.logger.error("Service account token couldn't be decoded, token is invalid")
+                decoded_token = None
+            except jwt.InvalidSignatureError:
+                app.logger.error("Service account token couldn't be decoded, signature is invalid")
+                decoded_token = None
+        else:
+            # For User account check if the audience is valid
+            for audience in audiences:
+                try:
+                    decoded_token = jwt.decode(token, public_key,
+                                               algorithms=['RS256'],
+                                               audience=audience)
+                except jwt.exceptions.InvalidAudienceError:
+                    app.logger.error("User auth token couldn't be decoded, audience is invalid")
+                    decoded_token = None
+                except jwt.InvalidTokenError:
+                    app.logger.error("User auth token couldn't be decoded, token is invalid")
+                    decoded_token = None
+                except jwt.InvalidSignatureError:
+                    app.logger.error("User auth token couldn't be decoded, signature is invalid")
+                    decoded_token = None
+
+            if decoded_token:
+                break
+
+    g.decoded_token = decoded_token or {}
+    return decoded_token
+
+
+def decode_user_token(app, token):
+    """Decode the authorization token read from the request header."""
+    audiences = get_audiences()
+    decoded_token = decode_token(app, token, audiences)
+
+    if decoded_token is None:
+        raise AuthError(401, 'Authentication failed - token missing or malformed')
+    if "email_verified" not in decoded_token:
+        raise jwt.InvalidIssuerError('Can not retrieve the email_verified property from the token')
+    if decoded_token["email_verified"] in ('0', 'False', 'false'):
+        raise jwt.InvalidIssuerError('Email of the user has not been validated')
+
+    return decoded_token
+
+
+def decode_service_token(app, token):
+    """Decode OSIO service token."""
+    decoded_token = decode_token(app, token)
+
+    return decoded_token
+
+
+def get_token_from_auth_header():
+    """Get the authorization token read from the request header."""
+    return request.headers.get('Authorization')
+
+
+def get_audiences():
+    """Retrieve all JWT audiences."""
+    return os.environ.get('FABRIC8_ANALYTICS_JWT_AUDIENCE', '').split(',')
+
+
+def login_required(view):
+    """Check if the login is required and if the user can be authorized."""
+    @wraps(view)
+    def wrapper(*args, **kwargs):
+        if os.environ.get('DISABLE_AUTHENTICATION') in ('1', 'True', 'true'):
```

Comment:
Nitpick: this code is used on more places, what about to refactor it to function like `is_authentication_disabled()`